pragma-node/src/infra/repositories/onchain_repository.rs

```diff
@@ -131,6 +134,114 @@ fn build_sql_query(
     Ok(complete_sql_query)
 }
 
+pub fn onchain_pair_exist(existing_pair_list: &Vec<EntryPairId>, pair_id: &str) -> bool {
+    existing_pair_list.iter().any(|entry| entry == pair_id)
+}
+
+fn calculate_rebased_price(
+    base_result: (BigDecimal, u32),
+    quote_result: (BigDecimal, u32),
+) -> Result<(BigDecimal, u32), InfraError> {
+    let (base_price, base_decimals) = base_result;
+    let (quote_price, quote_decimals) = quote_result;
+
+    if quote_price == BigDecimal::from(0) {
+        return Err(InfraError::InternalServerError);
+    }
+
+    let (rebase_price, decimals) = if base_decimals < quote_decimals {
+        let normalized_base_price =
+            normalize_to_decimals(base_price, base_decimals, quote_decimals);
+        (
+            convert_via_quote(normalized_base_price, quote_price, quote_decimals)?,
+            quote_decimals,
+        )
+    } else {
+        let normalized_quote_price =
+            normalize_to_decimals(quote_price, quote_decimals, base_decimals);
+        (
+            convert_via_quote(base_price, normalized_quote_price, base_decimals)?,
+            base_decimals,
+        )
+    };
+
+    Ok((rebase_price, decimals))
+}
+
+pub async fn routing(
+    onchain_pool: &Pool,
+    offchain_pool: &Pool,
+    network: Network,
+    pair_id: String,
+    timestamp: u64,
+    aggregation_mode: AggregationMode,
+    is_routing: bool,
+) -> Result<(BigDecimal, Vec<OnchainEntry>, Vec<String>, Option<u32>), InfraError> {
+    let existing_pair_list = get_existing_pairs(onchain_pool, network).await?;
+
+    if onchain_pair_exist(&existing_pair_list, &pair_id) || !is_routing {
```

Comment:
So here if the pair dors not exists on chain and the routing is False, we need to return a 404 error, ATM we return price 0 and an empty vec AFAIK